src/lib/_utilities/i18n.utilities.ts

```diff
@@ -109,7 +109,48 @@ export function interpolate(
     messageDirty = externalText;
   }
 
-  const messageClean = xss(messageDirty);
+  const messageClean = sanitize(messageDirty);
 
   return messageClean;
 }
+
+export function textToKey(text: string | number) {
```

Comment:
We could use Zod for this, but it feels like overkill. My main intention with it was to help enforce developer facing interfaces that are fed into the program by enforcing correct types both in build time and run time. I'm not using it much internally.

One thing to note is Zod's `parse` will `throw` if it fails. When we handle internal functionality like this, _if_ we did use Zod, I'd prefer `safeParse`, so it doesn't `throw`. With `safeParse`, we'd have to use a conditional to check for errors from the `safeParse`. So, with all things considered, it would likely complicate the function, rather than simplify.